homeassistant/components/august/sensor.py

```diff
@@ -0,0 +1,148 @@
+"""Support for August sensors."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.sensor import DEVICE_CLASS_BATTERY
+from homeassistant.helpers.entity import Entity
+
+from . import DATA_AUGUST
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=5)
+
+
+async def _async_retrieve_device_battery_state(detail):
+    """Get the latest state of the sensor."""
+    if detail is None:
+        return None
+
+    return detail.battery_level
+
+
+async def _async_retrieve_linked_keypad_battery_state(detail):
```

Comment:
why is this async ?

Reply:
It doesn't need to be anymore.  The original version did.  I'll refactor

Reply:
Fixed in branch battery_review_items_and_model

Reply:
https://github.com/home-assistant/home-assistant/pull/32187